core/string/ustring.cpp

```diff
@@ -4352,9 +4352,22 @@ bool String::is_valid_filename() const {
 		return false;
 	}
 
+	// TODO: Use invalid_filename_characters
 	return !(find(":") != -1 || find("/") != -1 || find("\\") != -1 || find("?") != -1 || find("*") != -1 || find("\"") != -1 || find("|") != -1 || find("%") != -1 || find("<") != -1 || find(">") != -1);
 }
 
+// Changes made to the set of invalid characters must also be reflected in the String documentation and the is_valid_filename function.
+static const char *invalid_filename_characters = ": / \\ ? * \" | % < >";
+
+String String::validate_filename() const {
+	Vector<String> chars = String(invalid_filename_characters).split(" ");
+	String name = this->replace(chars[0], "_");
+	for (int i = 1; i < chars.size(); i++) {
+		name = name.replace(chars[i], "_");
+	}
```

Comment:
Isn't this clearer?
```suggestion
	String name = this;
	for (int i = 0; i < chars.size(); i++) {
		name = name.replace(chars[i], "_");
	}
```

Reply:
Makes sense. I copied this from `validate_node_name` so I didn't have to think too hard :-p

Reply:
I forgot to put in the strip_edges(), so I changed it to `String name = strip_edges()`